src/psyclone/tests/psyir/backend/fortran_test.py

```diff
@@ -111,6 +112,115 @@ def test_gen_dims_error(monkeypatch):
     assert "unsupported gen_dims index 'invalid'" in str(excinfo)
 
 
+def test_gen_datatype():
+    '''Check the gen_datatype function produces the expected datatypes.'''
+    for datatype in ["real", "integer", "character"]:
```

Comment:
`@pytest.mark.parametrize`?

Reply:
Actually, you could fold this test into the following one?

Reply:
I've used parametrize. The test can't be folded into the next one as this test is without the optional precision argument and the next one is with it.